Library/test-helpers/lib.sh

```diff
@@ -1715,6 +1715,254 @@ limeSubmitCommonLogs() {
+
+Name of image tag, if paramer empty name tag set as agent_image.
+
+=back
+
+Returns 0.
+
+=cut
+
+limeconPrepareAgentImage() {
+
+    local IMAGE=$1
+    local NAME_TAG=$2
+
+    #copy cacert to work dir for podman build
+    cp /var/lib/keylime/cv_ca/cacert.crt .
+    
+    if podman images | grep agent_images; then
+        echo "This image is already builded!"
+        return 0
+    fi
+
+    if [ -z "${NAME_TAG}" ]; then
+        echo "Parameter specified name tag of build is empty. Setting default tag to: agent_image."
+        NAME_TAG="agent_image"
+    fi
+    sed -i '1s/^/FROM  '${IMAGE}' /' Dockerfile
+    podman build -t=$NAME_TAG .
+    echo "Recently builded image has tag: ${NAME_TAG}"
+    #clean working dir
+    rm -rf cacert.crt
+
+}
+
+true <<'=cut'
+=pod
+
+=head2 limeconRunAgentImage
+
+Container run via podman with specified parameters.
+
+    limeconRunAgentImage
+
+=over
+
+=item NAME_NETWORK
+
+Name of used podman network.
+
+=item NAME_CONTAINER
+
+Set name of container.
+
+=item AGENT_CONFIG
+
+Mounted dir with configuration file.
+
+=item TESTDIR
+
+Mounted test dir.
+
+=item IP_AGENT
+
+Ip address of container.
+
+=item IMAGE_TAG
+
+Name of image tag.
+
+=back
+
+Returns 0.
+
+=cut
+
+limeconRunAgentImage() {
+
+    local NAME_NETWORK=$1
+    local NAME_CONTAINER=$2
+    local AGENT_CONFIG=$3
+    local TESTDIR=$4
+    local IP_AGENT=$5
+    local IMAGE_TAG=$6
+
+    podman run -d --net $NAME_NETWORK --name $NAME_CONTAINER --privileged --volume=${AGENT_CONFIG}:/etc/keylime/ --volume=/sys/kernel/security/:/sys/kernel/security/:ro --volume=${TESTDIR}:${TESTDIR}:ro --ip ${IP_AGENT} --device=/dev/tpmrm0  localhost/$IMAGE_TAG /usr/bin/keylime_agent
+}
+
+true <<'=cut'
+=pod
+
+=head2 limeconSetupAgent
+
+Setup agent configuration for container and copy to confdir newly created confdir.
+
+    limeconSetupAgent
+
+=over
+
+=item AGENT_ID
+
+ID of keylime agent.
+
+=item AGENT_IP
+
+Ip address of keylime agent in container.
+
+=item AGENT_CONF
+
+Name of agent conf file.
+
+=item CONF_DIR
+
+Name of dir for agent config files.
+
+=back
+
+Returns 0.
+
+=cut
+
+limeconSetupAgent() {
+
+        local AGENT_ID=$1
+        local AGENT_IP=$2
+        local AGENT_CONF=$3
+        local CONF_DIR=$4
+
+        if [ -d "$CONF_DIR" ]; then
+            echo "Dir is already exist, copy file there."
+        else
+            mkdir -p $CONF_DIR
+        fi
+
+        limeUpdateConf agent uuid \"$AGENT_ID\"
+        limeUpdateConf agent ip \"$AGENT_IP\"
+        limeUpdateConf agent contact_ip \"$AGENT_IP\"
+        cp -r /etc/keylime $CONF_DIR${AGENT_CONF}
+
+}
+
+true <<'=cut'
+=pod
+
+=head2 limeconStopAgent
+
+Stop container and delete container.
+
+    limeconStopAgent
+
+=over
+
+=item CONT_NAME
+
+Root name of the containers to disable and delete.
+
+Returns 0.
+
+=cut
+
+limeconStopAgent() {
+
+        local CONT_NAME=$1
+
+        podman ps -a --format "{{.Names}}" | grep $CONT_NAME* | xargs podman stop | xargs podman rm     
```

Comment:
I am not sure if the output of `podman stop` can be input for `podman rm`.  Better split it into separate commands.
```
local NAMES
NAMES=$( podman ps -a --format "{{.Names}}" | grep $CONT_NAME* | xargs )
podman stop $NAMES
podman rm $NAMES
```

Reply:
In test seems that this approach also working. I found it [here](https://www.cloudbees.com/blog/docker-how-to-stop-and-remove-all-containers-at-once).

Reply:
OK. Still, `grep $CONT_NAME*` is not correct as it could match more than the user asked for.